_posts/2020-01-16-spring-bean-life-cycle.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "The Lifecycle of Spring Beans"
+categories: [spring-boot]
+modified: 2020-01-16
+excerpt: "An in-depth look at the paging support provided by Spring Data for querying
+          Spring Web MVC controllers and Spring Data repositories."
+image:
+  auto: 0012-pages
+
+---
+
+## What are Spring Beans?
+ > Spring Beans are Java Objects or Instances which is created and managed by Spring container.
+
+Spring `@ApplicationContext` is responsible for creating a Spring bean and manage the 
+Spring bean life cycle.
+
+Below, We have created a simple Spring bean for `Foo` class using `@Component` annotation.
+
+```java
+@Component
+Class Foo { //variables and methods ...}
+```
+## The Spring Bean Lifecycle
+
+The Bean's life cycle consists of 2 phases `Post-initialization` and `Pre-destruction`. 
+
+Below are the Phase 1, `Post-initialization` steps while creating a Bean.
+1. Bean Creation → using Annotations or XML Bean Configs.
+2. Bean Instantiate → In this step Bean will be loaded in `JVM` memory.
+3. Populating Bean properties →  Container will Create a Bean `id`, `scope`, `default values`.
+4. Implementing `BeanNameAware setBeanName()` → Getting to know the Bean Name created by Spring container.

```

Comment:
In the steps where you say "Implementing..." you're mixing things. In this section, we want to describe the lifecycle phases and not how to implement them. Can you think of another name you'd give to these lifecycle steps?